sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/datastore/DatastoreV1.java

```diff
@@ -1579,7 +1579,7 @@ public void finishBundle() throws Exception {
      * @throws DatastoreException if the commit fails or IOException or InterruptedException if
      *     backing off between retries fails.
      */
-    private void flushBatch() throws DatastoreException, IOException, InterruptedException {
+    synchronized private void flushBatch() throws DatastoreException, IOException, InterruptedException {
```

Comment:
As uniqueMutationKeys start working and flushBatch() is possible to get called, I am concerned about this apparently not thread-safe method. Within in bundle do we guarantee a processElement() being executed at the same time?